publish-to-npm/action.yml

```
@@ -0,0 +1,24 @@
+name: Publish to NPM
+
+author: Gabriel Montes
+
+description: Sets up Node, runs the prepublishOnly script and publishes the package to the NPM registry
+
+inputs:
+  token:
+    description: The NPM access token to use when publishing
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: hemilabs/actions/setup-node-env@main
+    - run: npm run --if-present prepublishOnly
+      shell: bash
+    - uses: JS-DevTools/npm-publish@v3
```

I think this action should be pinned to a commit hash, for both reliability and security reasons, especially since it is a third-party action.
```suggestion
    - uses: JS-DevTools/npm-publish@9ff4ebfbe48473265867fb9608c047e7995edfa3 # v3.1.1
```